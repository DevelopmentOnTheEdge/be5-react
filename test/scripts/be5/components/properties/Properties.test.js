import React from 'react';
import Properties from '../../../../../src/scripts/be5/components/properties/properties';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';

it('contain property', () => {
	const bean = require('./testJson.json');

	const component = renderer.create(
    <Properties bean={bean} ids={[0,1]}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('not contain property', () => {
    const bean = require('./testJson.json');

    const component = renderer.create(
        <Properties bean={bean} ids={[1000000]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


