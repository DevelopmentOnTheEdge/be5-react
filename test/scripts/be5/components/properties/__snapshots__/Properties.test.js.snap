// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contain property 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12"
  >
    <div
      className="form-group property required"
    >
      <label
        className="form-control-label"
        htmlFor="textInputField"
      >
        Text
      </label>
      <div
        className="controls"
      >
        <input
          className="form-control"
          disabled={undefined}
          id="textInputField"
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value="text"
        />
      </div>
    </div>
  </div>
  <div
    className="form-group property property-label col-md-12"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><strong>Pellentesque habitant morbi tristique</strong> senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. <em>Aenean ultricies mi vitae est.</em> Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, <code>commodo vitae</code>, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. <a href=\\"#\\">Donec non enim</a> in turpis pulvinar facilisis. Ut felis.</p><ol><li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li>   <li>Aliquam tincidunt mauris eu risus.</li></ol>",
        }
      }
    />
  </div>
</div>
`;

exports[`not contain property 1`] = `
<div
  className="row"
/>
`;
